sklearn/utils/tests/test_validation.py

```diff
@@ -975,23 +977,54 @@ def test_check_is_fitted_with_attributes(wrap):
         check_is_fitted(ard, wrap(["coef_bad_"]))
 
 
-def test_check_consistent_length():
-    check_consistent_length([1], [2], [3], [4], [5])
-    check_consistent_length([[1, 2], [[1, 2]]], [1, 2], ["a", "b"])
+def test_check_consistent_length_sparse():
+    """Test that `check_consistent_length` works with sparse data."""
     check_consistent_length([1], (2,), np.array([3]), sp.csr_matrix((1, 2)))
+
+
+@pytest.mark.parametrize(
+    "array_namespace, device, dtype_name", yield_namespace_device_dtype_combinations()
+)
+def test_check_consistent_length(array_namespace, device, dtype_name):
+    """Test that `check_consistent_length` raises on inconsistent lengthes and wrong
+    input types trigger TypeErrors."""
+    xp = _array_api_for_tests(array_namespace, device)
+
+    check_consistent_length(
+        xp.asarray([1], device=device),
+        xp.asarray([2], device=device),
+        xp.asarray([3], device=device),
+        xp.asarray([4], device=device),
+        xp.asarray([5], device=device),
+    )
+    if xp.__name__ == "numpy":
+        check_consistent_length(
+            xp.asarray([[1, 2], [1, 2]], device=device),
```

Comment:
Here, I had to change the input into the test, because it would raise on numpy and torch. It was [[1, 2], [[1, 2]]] originally.
While `check_consistent_length` only checks the length of the data (length along axis 0), but those libraries would raise as well, when the shape along axis 1 would not be consistent.

Reply:
I'm not sure if I understand, could you please write what exactly would raise?

Reply:
Sorry, I will try to explain in more detail:

`xp.asarray([[1, 2], [[1, 2]]], device=device)` (with the two extra brackets) will raise both on numpy and on torch, because both libraries cannot create arrays with inconsistent shapes. 

The numpy error:
`ValueError: setting an array element with a sequence. The requested array has an inhomogeneous shape after 1 dimensions. The detected shape was (2,) + inhomogeneous part.`

The torch error:
`ValueError: expected sequence of length 2 at dim 1 (got 1)`

Our `check_consistent_length` is more flexible, at least when we pass python lists to it.


Note: The `if xp.__name__ == "numpy"` distinction is unrelated to this. Is is only for being able to pass strings into the creation of an array, which numpy tolerates, but torch does not.

Reply:
It's only a side comment, in case I oversaw the original reason to craft the input to the test like it was done.
I don't think that it has any effect changing the test like this.